### PR TITLE
Fix ClassCastException when querying object array within object array

### PR DESCRIPTION
**Pull request by jeeminso**
## Summary of the changes / Why this improves CrateDB
Fixes:
```
cr> create table t (o object as (o2 object as (a int)[])[]);
CREATE OK, 1 row affected (1.583 sec)
cr> insert into t values ([ {o2 = [ {a=1} ] } ]);
INSERT OK, 1 row affected (0.152 sec)
cr> select o['o2'] from t;
ClassCastException[class java.util.ArrayList cannot be cast to class java.util.Map (java.util.ArrayList and java.util.Map are in module java.base of loader 'bootstrap')]
```
The ClassCastException is thrown from `ColumnFieldVisitor.getDocMap`(which is a `StoredFieldVisitor`) while attempting to merge the stored fields, `o` and `o['o2']`. But see the docs for `IndexDocumentBuilder.noStoredField()`, `o['o2']` must not need to be stored as it is a child of an object array. What caused `o['o2']` to be stored was `IndexDocumentBuilder.wrapTranslog()` method which failed to preserve the state that `noStoredField()` was previously called.

## Checklist

 - [x] Added an entry in the latest `docs/appendices/release-notes/<x.y.0>.rst` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in the latest `docs/appendices/release-notes/<x.y.0>.rst`
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

**Comment by jeeminso** [comment]
Works on `5.9.2`:
```
cr> create table t (o object as (o2 object as (a int)[])[]);
CREATE OK, 1 row affected (0.307 sec)
cr> insert into t values ([ {o2 = [ {a=1} ] } ]);
INSERT OK, 1 row affected (0.093 sec)
cr> select o['o2'] from t;
+--------------+
| o['o2']      |
+--------------+
| [[{"a": 1}]] |
+--------------+
SELECT 1 row in set (0.079 sec)
```

**Comment by BaurzhanSakhariev** [comment]
>Works on 5.9.2:

Is it a regression on master (and on 5.9.3)? If yes, what was the commit causing it?

**Comment by romseygeek** [comment]
> Is it a regression on master (and on 5.9.3)? If yes, what was the commit causing it?

It's a regression on master only; it's caused by the per-column storage changes.

File: server/src/main/java/io/crate/execution/dml/IndexDocumentBuilder.java [hunk 1]
```diff
@@ -61,21 +62,30 @@ public IndexDocumentBuilder(
...
     }
 
     private IndexDocumentBuilder(
         Document doc,
         TranslogWriter translogWriter,
         ValueIndexer.Synthetics synthetics,
         Map<ColumnIdent, Indexer.ColumnConstraint> constraints,
-        Version tableVersionCreated
+        Version tableVersionCreated,
+        Boolean addStoredField
```
Comment (romseygeek) [inline server/src/main/java/io/crate/execution/dml/IndexDocumentBuilder.java:81]:
```suggestion
        boolean addStoredField
```

**Review by romseygeek — APPROVED** [review]
LGTM, thanks @jeeminso

**Pull request merged**